### DISCUSSION TITLE
reusable components

### DISCUSSION THREAD

**Discussion by rito** [Budibase/budibase#7684] [Ideas]
I haven't found a simple way to reuse components.

There are several components on a page that are reusable, either in layout or functionality.
For example:
personalized menus,
footers
cards
Graphics
Popup Modal
In addition to graphical items, some automations and validations can be reusable, for example, when entering the screen Running X function. Or a validation of a form field that also exists on several other screens.
Rather than creating and duplicating the same functions and components, these could be reusable. Do it once and reuse it on the other screens. After all, if I have 50 screens and out of 50 I need the same component, I need to create 50 times and in an update, do it 50 times more.

**Comment by melohagan** [comment]
More templating and reuse options have been requested before; it would be nice to have!
